src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -847,42 +848,134 @@ private void RebuildRegionConnections()
             var region2 = map.Regions[entry.Key.y];
 
-            var region1Shown = region1.Visibility == MapElementVisibility.Shown;
-            var region2Shown = region2.Visibility == MapElementVisibility.Shown;
+            var vis1 = region1.Visibility;
+            var vis2 = region2.Visibility;
 
-            if (!region1Shown && !region2Shown)
-            {
+            // Do not draw connections between hidden or unknown regions
+            if (vis1 != MapElementVisibility.Shown && vis2 != MapElementVisibility.Shown)
                 continue;
-            }
 
-            var points = entry.Value;
-            var highlight = CheckHighlightedAdjacency(region1, region2);
+            // Check if connections should be highlighted
+            // (Unknown patches should not highlight connections)
+            var highlight = CheckHighlightedAdjacency(region1, region2) &&
+                SelectedPatch?.Visibility == MapElementVisibility.Shown;
+
             var color = highlight ? HighlightedConnectionColor : DefaultConnectionColor;
 
-            var line = CreateRegionConnectionLine(points, color);
+            // Create the main connection line
+            var points = entry.Value;
+            var line = CreateConnectionLine(points, color);
             regionConnectionLines.Add(entry.Key, line);
 
-            if (region1Shown && !region2Shown)
+            // Fade the connection line if need be
+            ApplyFadeIfNeeded(region1, region2, line, false);
+            ApplyFadeIfNeeded(region2, region1, line, true);
+
+            // Create additional lines to connect "floating" patches in unknown regions
+            if (vis1 == MapElementVisibility.Unknown)
             {
-                ApplyFadeToLine(line, false);
+                additionalConnectionLines
+                    .AddRange(BuildUnknownRegionConnections(line, region1, region2, color, false));
             }
-            else if (!region1Shown && region2Shown)
+
+            if (vis2 == MapElementVisibility.Unknown)
+            {
+                additionalConnectionLines
+                    .AddRange(BuildUnknownRegionConnections(line, region2, region1, color, true));
+            }
+        }
+    }
+
+    private Line2D[] BuildUnknownRegionConnections(Line2D startingConnection, PatchRegion targetRegion,
+        PatchRegion startRegion, Color color, bool reversed)
+    {
+        var startingPoint = reversed ?
+            startingConnection.Points[startingConnection.Points.Length - 1] :
+            startingConnection.Points[0];
+
+        var adjacencies = startRegion.PatchAdjacencies[targetRegion.ID];
+
+        // Generate a list of patches to connect to
+        var patches = targetRegion.Patches
+            .Where(p => p.Visibility == MapElementVisibility.Unknown)
+            .Where(p => adjacencies.Contains(p))
+            .ToArray();
+
+        var newConnections = new Line2D[patches.Length];
```

Comment:
I'd prefer if the newConnections was a List so that the ToArray call can be avoided here. Or even better I think this whole method could be made into an iterator (https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/yield).